src/PureSMT/SExpr.hs

```diff
@@ -62,14 +62,26 @@ overAtomS :: (String -> String) -> SExpr -> SExpr
 overAtomS f (Atom s) = Atom (f s)
 overAtomS f (List ss) = List [overAtomS f s | s <- ss]
 
--- | Convert an s-expression to a (strict) null terminated bytestring.
-serializeSExpr :: SExpr -> BS.ByteString
-serializeSExpr = LBS.toStrict . toLazyByteString . (<> "\NUL") . renderSExpr
-  where
-    renderSExpr :: SExpr -> Builder
-    renderSExpr (Atom x) = stringUtf8 x
-    renderSExpr (List es) =
-      "(" <> mconcat (intersperse " " [renderSExpr e | e <- es]) <> ")"
+-- | Evaluate a bytestring builder to a strict bytestring that is expected to be
+-- consumed immediately.
+serializeUntrimmed :: Int -> Int -> Builder -> BS.ByteString
+serializeUntrimmed firstChunkSize newChunksSize = LBS.toStrict . toLazyByteStringWith (untrimmedStrategy firstChunkSize newChunksSize) "\NUL"
+
+-- | Evaluate a bytestring builder corresponding to a single SMTLib2 command
+-- (the size of the buffer is expected to be small).
+serializeSingle :: Builder -> BS.ByteString
+serializeSingle = serializeUntrimmed 256 2048
```

Comment:
256 is the first power of 2 that is bigger than the length of the longest command with interesting output in 'isUnity', and 2048 is just four times that because `smallChunkSize * 4 = defaultChunkSize`.

Reply:
Thanks. Would be good to put it in a comment too.